### PR TITLE
Do not panic if sending response signal fails

### LINKED ISSUE #1002: Crash after x hours running during testing autopeering
## Bug description
Testing the autopeering. After 5 hours running the node crashes
## Rust version
Which version of Rust are you running?
- Rust version:
` stable-x86_64-pc-windows-msvc updated - rustc 1.58.1 (db9d1b20b 2022-01-20) (from rustc 1.58.0 (02072b482 2022-01-11))`
## Which version of Bee are you using?
```
PS D:\github\bee\bee-node> ..\target\release\bee.exe -v
0.3.0-rc2-99682d7
PS D:\github\bee\bee-node>
```

## Hardware specification

What hardware are you using?

- Operating system: W10 Prof
- RAM: 32 GB
- Cores: 48 AMD Threadripper 3960X
- Device: SSD

## Steps To reproduce the bug

1. Enable autopeering
2.  When needed add 1 static peer to get synced
3. After x hours the node crashes. Seen it twice now, once after 12 hours, other after 5 hours.

## Expected behaviour
Node should be finding enough peers to keep synced

## Actual behaviour

Crash after x hours.

## Errors
`tokio-runtime-worker' panicked at 'error sending response signal: [10, 32, 133, 47, 198, 27, 9, 63, 90, 178, 253, 236, 228, 11, 47, 209, 146, 16, 186, 3, 121, 180, 161, 205, 112, 192, 118, 49, 238, 23, 181, 98, 5, 215, 18, 52, 10, 19, 10, 7, 112, 101, 101, 114, 105, 110, 103, 18, 8, 10, 3, 117, 100, 112, 16, 162, 114, 10, 29, 10, 17, 99, 104, 114, 121, 115, 97, 108, 105, 115, 45, 109, 97, 105, 110, 110, 101, 116, 18, 8, 10, 3, 116, 99, 112, 16, 240, 121, 26, 12, 57, 52, 46, 50, 49, 50, 46, 51, 49, 46, 50, 52]', C:\Users\user\.cargo\git\checkouts\bee-781893e298f3299f\10914be\bee-network\bee-autopeering\src\discovery\manager.rs:686:14
`
Last ~700 lines from debug-logging and actual crash in attached logfile.
config.toml also attached as .txt
[bee_crash_autopeering_testing.log](https://github.com/iotaledger/bee/files/7918393/bee_crash_autopeering_testing.log)
[config_anon.toml.txt](https://github.com/iotaledger/bee/files/7918410/config_anon.toml.txt)

### PR DESCRIPTION
**Pull request by Alex6323**
# Description of change

Bug fix.

## Links to any relevant issues

Probably fixes #1002 

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

File: bee-network/bee-autopeering/src/discovery/manager.rs [hunk 1]
```diff
@@ -681,9 +681,9 @@ fn handle_verification_response(verif_res: VerificationResponse, verif_reqval: R
 
     // Send the response notification.
     if let Some(tx) = verif_reqval.response_tx {
-        // Panic: we don't allow channel send errors.
-        tx.send(verif_res.to_protobuf().to_vec())
-            .expect("error sending response signal");
+        // Note: if sending fails, then the response arrived a tiny bit too late for the predefined timeout (race
+        // condition), so we just ignore the error in that case.
+        let _ = tx.send(verif_res.to_protobuf().to_vec());
```
Comment (grtlr) [inline bee-network/bee-autopeering/src/discovery/manager.rs]: Maybe we should emit a `warn` in those three cases?

Comment (Alex6323) [reply bee-network/bee-autopeering/src/discovery/manager.rs]: Would `debug` also be fine for you? `warn` would indicate that the node could do something about it but it really can't.

Comment (grtlr) [reply bee-network/bee-autopeering/src/discovery/manager.rs]: Yeah, that's right. `debug` is fine with me 😊

Comment (Alex6323) [reply bee-network/bee-autopeering/src/discovery/manager.rs]: Done

**Review by thibault-martinez — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**